### ISSUE TITLE
Can't get localized terms

### ISSUE THREAD

**Issue by aerni** [statamic/cms#3274]
## Bug Description
I have multiple Taxonomies and need to get all Terms. I'm expecting the following to return all the Terms, similar to how you can get all Entries with the `Entry` Facade.

```php
return Term::all();
```

```php
```

## How to Reproduce
- Create a multi-site with two sites
- Create a taxonomy and add both sites to it
- Customize the taxonomy blueprint to have at least one localizable field
- Create a term
- Localize the term
- Try to get the values

## Environment

**Statamic version:** 3.0.{?}

**PHP version:** 7.{?}

**Install method (choose one):**
- Fresh install from `statamic/statamic`
- Starter kit
- Existing Laravel app
- Other (please explain this magic!)

**Comment by duncanmcclean** [comment]
So instead of [`LocalizedTerm`](https://github.com/statamic/cms/blob/3.0/src/Taxonomies/LocalizedTerm.php) objects, you're expecting it to return [`Term`](https://github.com/statamic/cms/blob/3.0/src/Taxonomies/Term.php) objects?

**Comment by aerni** [comment]
Not exactly sure. I think a `LocalizedTerm` is fine. What I'm expecting is an object for each localization of each Term. I have a Taxonomy with three Terms. Each Term is localized in `English` and `German`. So I'm expecting a total of six `LocalizedTerm` objects. 

This here:
```php
return Term::all()->map(function ($term) {
    dd($term);
});
```

Dumps this:
```php
Statamic\Taxonomies\LocalizedTerm {#2760
  #locale: "english"
  #term: Statamic\Taxonomies\Term {#2735
    #taxonomy: "categories"
    #slug: "clothes"
    #blueprint: "categories"
    #collection: null
    #data: Illuminate\Support\Collection {#2731
      #items: array:2 [
        "german" => Illuminate\Support\Collection {#2737
          #items: array:5 [
            "title" => "Kleider"
            "slug" => "kleider"
            "updated_by" => "59e57af6-a970-4544-a83f-7b51e30e1665"
            "updated_at" => 1613653762
            "exclude_from_static_cache" => true
          ]
        }
        "english" => Illuminate\Support\Collection {#2779
          #items: array:5 [
            "title" => "Clothes"
            "updated_by" => "59e57af6-a970-4544-a83f-7b51e30e1665"
            "updated_at" => 1613648108
            "blueprint" => "categories"
            "exclude_from_static_cache" => true
          ]
        }
      ]
    }
    #initialPath: "/Users/michael/Code/statamic-dynamic-cache/content/taxonomies/categories/clothes.yaml"
  }
  #slug: null
  #published: true
  #selectedQueryColumns: null
  #supplements: Illuminate\Support\Collection {#2839
    #items: []
  }
}
```

The `LocalizedTerm` object includes all Term localizations rather than only one localization at a time. So there's no to get the data of each localization separately, the same way you do with Collection Entries.

With Collection Entries you can do `$entry->values()`, which gives you the origin Entry merged with the localized Entry. I'm expecting Taxonomy Terms to work the same way.

Long story short, I need a way to get all localizations of all Terms.

**Comment by mattyybee** [comment]
Yeah, this needs to be a thing. Had to do the same ugly stuff to manually get any localized terms.

**Comment by jasonvarga** [comment]
In 3.4, `Term::all()`, `Term::query()->get()`, etc will return _all_ localizations. #6318
